client/src/main/java/io/pravega/client/state/impl/RevisionedStreamClientImpl.java

```diff
@@ -260,4 +260,25 @@ void handleSegmentSealed() {
          .filter(pendingEvent -> pendingEvent.getAckFuture() != null)
          .forEach(pendingEvent -> pendingEvent.getAckFuture().completeExceptionally(new SegmentSealedException(segment.toString())));
     }
+
+    public Iterator<Entry<Revision, T>> readRange(Revision startRevision, Revision endRevision) {
+        log.trace("Read segment {} from revision {} to revision {}", segment, startRevision, endRevision);
+        long tempOffset;
+        synchronized (lock) {
+            long startOffset = startRevision.asImpl().getOffsetInSegment();
+            SegmentInfo segmentInfo = Futures.getThrowingException(meta.getSegmentInfo());
+            long endOffset = endRevision.asImpl().getOffsetInSegment();
+            if (startOffset < segmentInfo.getStartingOffset()) {
+                throw new TruncatedDataException(format("Data at the supplied revision {%s} has been truncated. The current segment info is {%s}", startRevision, segmentInfo));
+            }
+            if (startOffset == endOffset) {
+                log.debug("No new updates to be read from revision {}", endRevision);
+            }
+            if (startOffset > endOffset) {
+                throw new IllegalStateException("startOffset {} is grater than endOffset {}");
+            }
```

Comment:
what happens if given end revision is greater than current write position in stream?